test/model/src/core/kv_table.cpp

```diff
@@ -227,6 +227,59 @@ kv_table::remove(kv_transaction_ptr txn, const data_value &key)
     }
 }
 
+/*
+ * kv_table::truncate --
+ *     Truncate a key range.
+ */
+int
+kv_table::truncate(const data_value &start, const data_value &stop, timestamp_t timestamp)
```

Comment:
I've been keeping both sets of APIs, but I'm increasingly getting the conviction that the non-transactional versions are not really necessary apart from convenience in testing. The next few tickets will determine what's really needed here ([WT-11620](https://jira.mongodb.org/browse/WT-11620) and [WT-11621](https://jira.mongodb.org/browse/WT-11621)), but I'm pretty sure that I'll be using only the transactional APIs there.

Even if we do end up keeping them, I'm thinking that they should be just wrappers around the APIs that take the transaction object, but I would rather to that in a separate ticket to avoid scope creep here?

Reply:
Ack - happy to leave it for later.

Reply:
I just created a ticket [WT-12146](https://jira.mongodb.org/browse/WT-12146) that proposes to re-implement the non-transactional API as a wrapper around the transactional API.